src/main/java/net/neoforged/neoforge/event/entity/EntityInvulnerablityCheckEvent.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity;
+
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired when {@link Entity#hurt(DamageSource, float)} is invoked and determines if
+ * downstream hurt logic should apply. This event is fired in {@link Entity#isInvulnerableTo(DamageSource)}
+ * <p>
+ * This event cannot be cancelled.
+ * <p>
+ * This event does not have a result.
+ * <p>
+ * This event is fired on the {@link NeoForge#EVENT_BUS}.
```

Comment:
These lines are not necessary, they are implicit from the event's class declaration (old events have them due to the old annotation system).

In their place I find it is most applicable to put a line noting which logical side the event will be fired on.